### PR TITLE
Show state data in a screen

### PR DESCRIPTION
**Pull request by danielsousaio**

### CODE REVIEW

File: src/features/BlankScreen011967/screens/blank.js [hunk 1]
```diff
@@ -10,54 +10,65 @@ import {
...
+    alignItems: 'center',
+    backgroundColor: '#DDDDDD',
+    padding: 10,
+  },
 });
 function mapStateToProps(state) {
-  return { state: state };
+  return {
+    users: state.apiReducer.users,
+    registration: state.apiReducer.registration,
```
Comment (danielsousaio) [inline src/features/BlankScreen011967/screens/blank.js:60]: This gives access to part of the state tree to this screen. Notice the reducer consequents return new objects for those keys (through `Object.assign`)

File: src/features/BlankScreen111968/screens/blank.js [hunk 2]
```diff
@@ -83,26 +83,25 @@ const styles = StyleSheet.create({
...
+    alignContent: 'stretch',
+    borderColor: '#000000',
+    borderStyle: 'solid',
     borderWidth: 0,
     borderLeftWidth: 0,
     borderRightWidth: 0,
     borderTopWidth: 0,
     borderBottomWidth: 0,
     borderRadius: 0,
-    backgroundSize: "auto"
```
Comment (danielsousaio) [inline src/features/BlankScreen111968/screens/blank.js:105]: Invalid CSS property, wouldn't compile

File: src/store/reducers.js [hunk 3]
```diff
@@ -1,159 +1,164 @@
...
     case types.REGISTRATION_POST_RESTAUTH_REGISTRATION_CREATE_SUCCEEDED:
+      return Object.assign({}, state, {
+        users: [...state.users, action.response],
+      });
     case types.REGISTRATION_POST_RESTAUTH_REGISTRATION_CREATE_FAILED:
       return Object.assign({}, state, {
-        registration: [...state.registration, action.response]
-      })
+        registration: action.error,
+      });
```
Comment (danielsousaio) [inline src/store/reducers.js:34]: Added return inside the success switch consequent

File: src/store/sagas.js [hunk 4]
```diff
@@ -1,338 +1,346 @@
-import { put, call, all, spawn, takeEvery } from "redux-saga/effects"
-import { apiService } from "./services"
-import * as types from "./constants"
+import {put, call, all, spawn, takeEvery} from 'redux-saga/effects';
+import {apiService} from './services';
+import * as actions from './actions';
```
Comment (danielsousaio) [inline src/store/sagas.js:3]: Missing import, added.

**Review by danielsousaio — COMMENTED** [review]
Changes summary above

File: src/features/BlankScreen011967/screens/blank.js [hunk 5]
```diff
@@ -10,54 +10,65 @@ import {
...
         style={styles.button}
-        onPress={() => this.props.signup(this.state.email, this.state.password)}
-      >
+        onPress={() =>
+          this.props.signup(this.state.email, this.state.password)
+        }>
         <Text>Signup</Text>
       </TouchableOpacity>
+      <Text>User count {this.props.users.length}</Text>
+      <Text>Errors {this.props.registration.message}</Text>
```
Comment (danielsousaio) [inline src/features/BlankScreen011967/screens/blank.js:42]: Show the state data

File: src/store/reducers.js (hunk 3 above)
Comment (charathram) [reply src/store/reducers.js:34]: Could you please open a PR to fix this in the codegen?

File: src/store/sagas.js (hunk 4 above)
Comment (charathram) [reply src/store/sagas.js:3]: Could you please open a PR to fix this in codegen?

File: src/store/reducers.js (hunk 3 above)
Comment (charathram) [reply src/store/reducers.js:34]: @danielsousaio Re: Line 29... I'm assuming that `users` will include the entire success response from the API. Correct? So, to access the API token that's returned on a successful signup (which comes via a `key` property in the response) I'd do something like `users.key`. Is that correct?

File: src/store/sagas.js (hunk 4 above)
Comment (danielsousaio) [reply src/store/sagas.js:3]: Done: https://github.com/crowdbotics/react-native-scaffold/pull/215

File: src/store/reducers.js (hunk 3 above)
Comment (danielsousaio) [reply src/store/reducers.js:34]: Yes, but first you would look for the user record within the array.
```
users.filter(user => user.id !== <id_you're_looking_for>).key
```

Comment (danielsousaio) [reply src/store/reducers.js:34]: The above lookup is usually abstracted with redux selectors

**Pull request closed without merging**